src/guidellm/config/base.py

```diff
@@ -0,0 +1,113 @@
+from typing import Optional
+from enum import Enum
+
+from pydantic import BaseModel, model_validator
+from pydantic_settings import BaseSettings, SettingsConfigDict
+
+
+__all__ = [
+    "settings",
+    "Settings",
+    "Environment",
+    "LoggingSettings",
+    "OpenAISettings",
+    "ReportGenerationSettings",
+]
+
+
+class Environment(str, Enum):
+    """
+    Enum for the supported environments
+    """
+
+    LOCAL = "local"
+    DEV = "dev"
+    STAGING = "staging"
+    PROD = "prod"
```

Comment:
What about [this approach](https://github.com/neuralmagic/guidellm/blob/e1ec99ac85ac9f6a7ee4b497ccaaf227a119889d/src/guidellm/core/serializable.py#L13) here?

Reply:
Need to add that approach back in, and I think it works well for that use case, but with Pydantic we don't need to deal with the constructors and it auto adds the choices to make sure they match rather than us adding some more configuration to a field